src/main/java/com/sourcegraph/cody/agent/protocol/TelemetryEventParameters.kt

```diff
@@ -1,8 +1,20 @@
 package com.sourcegraph.cody.agent.protocol
 
+enum class BillingProduct {
+  CODY
+}
+
+enum class BillingCategory {
+  CORE,
+  BILLABLE
+}
+
+data class BillingMetadata(val product: BillingProduct, val category: BillingCategory)
+
 data class TelemetryEventParameters(
```

Comment:
btw, this protocol class is not generated 
```
interface TelemetryEvent {
    feature: string
    action: string
    parameters?:
        | TelemetryEventParameters<{ [key: string]: number }, BillingProduct, BillingCategory>
        | undefined
        | null
}
```
It looks like our generating tool does not support fields like `parameters` in this case. I'd be nice to get it working.